cli/azd/test/functional/init_test.go

```diff
@@ -291,6 +291,56 @@ func Test_CLI_Init_CanUseTemplate(t *testing.T) {
 	require.FileExists(t, filepath.Join(dir, "README.md"))
 }
 
+// Test_CLI_Init_WithCwdAutoCreate tests the automatic directory creation when using -C/--cwd flag.
+func Test_CLI_Init_WithCwdAutoCreate(t *testing.T) {
+	tests := []struct {
+		name   string
+		subDir string // subdirectory to create within temp dir (using -C flag)
+		args   []string
+	}{
+		{
+			name:   "single level directory",
+			subDir: "new-project",
+			args:   []string{"init", "-t", "azure-samples/todo-nodejs-mongo", "--no-prompt"},
+		},
+		{
+			name:   "nested directory",
+			subDir: "parent/child/project",
+			args:   []string{"init", "-t", "azure-samples/todo-nodejs-mongo", "--no-prompt"},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx, cancel := newTestContext(t)
+			defer cancel()
+
+			// Create a parent temp directory
+			parentDir := tempDirWithDiagnostics(t)
+			cli := azdcli.NewCLI(t)
+			cli.WorkingDirectory = parentDir
+
+			// Add -C flag with the subdirectory path
+			targetDir := filepath.Join(parentDir, tt.subDir)
+			args := append([]string{"-C", tt.subDir}, tt.args...)
```

Comment:
@vhvb1989 Does this work? I didn't know we supported a `-C` flag. I was under the impression that `--cwd` is the only option currently available.